xwiki-platform-core/xwiki-platform-oldcore/src/main/java/com/xpn/xwiki/doc/XWikiDocument.java

```diff
@@ -4846,17 +4832,14 @@ public XWikiAttachment removeAttachment(XWikiAttachment attachment)
     public XWikiAttachment removeAttachment(XWikiAttachment attachmentToRemove, boolean toRecycleBin)
     {
         List<XWikiAttachment> list = getAttachmentList();
-        for (int i = 0; i < list.size(); i++) {
-            XWikiAttachment attachment = list.get(i);
-            if (attachmentToRemove.getFilename().equals(attachment.getFilename())) {
-                list.remove(i);
-                this.attachmentsToRemove.add(new XWikiAttachmentToRemove(attachment, toRecycleBin));
-                setMetaDataDirty(true);
-                return attachment;
-            }
+        if (!list.contains(attachmentToRemove)) {
```

Comment:
You don't need this if. You could simply rely on list.remove return (which is true when it found something to remove).